src/components/structures/TimelinePanel.tsx

```diff
@@ -806,23 +806,34 @@ class TimelinePanel extends React.Component<IProps, IState> {
         // Can be null for the notification timeline, etc.
         if (!this.props.timelineSet.room) return;
 
+        if (ev.getRoomId() !== this.props.timelineSet.room.roomId) return;
+
+        if (!this.state.events.includes(ev)) return;
+
+        this.recheckFirstVisibleEventIndex();
+
         // Need to update as we don't display event tiles for events that
         // haven't yet been decrypted. The event will have just been updated
         // in place so we just need to re-render.
         // TODO: We should restrict this to only events in our timeline,
         // but possibly the event tile itself should just update when this
         // happens to save us re-rendering the whole timeline.
-        if (ev.getRoomId() === this.props.timelineSet.room.roomId) {
-            this.buildCallEventGroupers(this.state.events);
-            this.forceUpdate();
-        }
+        this.buildCallEventGroupers(this.state.events);
+        this.forceUpdate();
     };
```

Comment:
Is it possible to write a test for this scenario so we don't regress? This stuff seems very order dependent and tricky to get right

Reply:
We can regression test the single edge we identified and fixed here, but there are so many edges, you will struggle to cover them all.

Reply:
I think it should be possible - I'll have a look into it, would you mind creating an issue?

Reply:
https://github.com/vector-im/element-web/issues/22155 ⏩